packages/back-end/src/services/features.ts

```diff
@@ -1238,32 +1216,27 @@ export function applyExperimentHashing(
   });
 }
 
-// Specific hashing entrypoint for idList values
+// Specific hashing entrypoint for SavedGroup objects
 export function applySavedGroupHashing(
-  savedGroups: SavedGroupsValues,
+  savedGroups: SavedGroupInterface[],
   attributes: SDKAttributeSchema,
-  savedGroupAttributeKeys: Record<string, string>,
   salt: string
-): SavedGroupsValues {
-  return Object.fromEntries(
-    Object.entries(savedGroups).map(([savedGroupId, savedGroupItems]) => {
-      const attribute = attributes.find(
-        (attr) => attr.property === savedGroupAttributeKeys[savedGroupId]
-      );
-      return attribute
-        ? [
-            savedGroupId,
-            hashStrings({
-              obj: savedGroupItems,
-              salt,
-              attributes,
-              attribute,
-              doHash: attribute.hashAttribute,
-            }),
-          ]
-        : [savedGroupId, savedGroupItems];
-    })
-  );
+): SavedGroupInterface[] {
+  return savedGroups.map((group) => {
+    const attribute = attributes.find(
+      (attr) => attr.property === group.attributeKey
+    );
+    if (attribute) {
+      group.values = hashStrings({
```

Comment:
Instead of mutating the group, should we shallow clone first?

Reply:
Ah yeah that's in a map huh. I'm gonna add at least one unit test so I'll put that in with it as well.
Not sure how this bug got missed when I explicitly tested this behavior. Must've changed just before final merge or after some of the patching.